release.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -e
+
+[ $# == 1 ] || { echo "usage: $0 version" && exit 1; }
+
+VER=$1
+
+[[ "$VER" =~ ^v[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+$ ]] || {
+	echo "malformed version: \"$VER\""
+	exit 2
+}
+
+if test -n "$(git ls-files --others | \
+	grep --invert-match '\(vendor\|build/operator-sdk-v.\+\)')";
+then
+	echo "directory has untracked files"
+	exit 1
+fi
+
+if ! $(git diff-index --quiet HEAD --); then
+	echo "directory has uncommitted files"
+	exit 1
+fi
+
+# Detect whether versions in code were updated.
+CURR_VER="$(git describe --dirty --tags)"
+VER_FILE="version/version.go"
+TOML_TMPL_FILE="pkg/generator/templates.go"
+CURR_VER_VER_FILE="$(sed -nr 's/Version = "(.+)"/\1/p' "$VER_FILE" | tr -d '\s\t\n')"
+CURR_VER_TMPL_FILE="$(sed -nr 's/.*".*v(.+)".*#osdk_version_annotation/v\1/p' "$TOML_TMPL_FILE" | tr -d '\s\t\n')"
+if [ "$CURR_VER" != "$CURR_VER_VER_FILE" ] \
```

Comment:
This should be comparing the desired version with verison.go. `if [ "$VER" != "$CURR_VER_VER_FILE" ]`
Say I'm trying to cut the tag for v0.0.7 and I've updated version.go and Gopkg.toml to v0.0.7.

But before this script can actually cut the tag, my current tag would be v0.0.6.
The current tag is something the script would update. The check should be to make sure version.go and Gopkg.toml are already updated to use the desired `$VER`.